src/JsonTxBuilderBase.sol

```diff
@@ -57,6 +57,19 @@ abstract contract JsonTxBuilderBase is CommonBase {
     ///         This function can be overridden to provide more specific checks, but should still call
     ///         super.checkStateDiff().
     function checkStateDiff(Vm.AccountAccess[] memory accountAccesses) internal view virtual {
+        address[] memory empty = new address[](0);
+        checkStateDiff(accountAccesses, empty);
+    }
+
+    /// @notice Reads all account and storage accesses and makes a series of basic sanity checks on them.
+    ///         This function can be overridden to provide more specific checks, but should still call
+    ///         super.checkStateDiff().
+    /// @param  allowedAccesses - if specified, it is checked for each memory access
+    ///         that the account is present in this array.
+    function checkStateDiff(Vm.AccountAccess[] memory accountAccesses, address[] memory allowedAccesses)
```

Comment:
We should instead implement this the same way we do `getCodeExceptions`—this approach would be a breaking change to #232 because of the change to the `super.checkStateDiff(accountAccesses)` signature

Concretely, add a method virtial method called `getAccountsWithExpectedStorageWrites` (or some better, less verbose name) that has a default implementation similar to `getCodeExceptions`, then:

```solidity
 function checkStateDiff(Vm.AccountAccess[] memory accountAccesses) internal view virtual {
    address[] memory allowedAccesses = getAccountsWithExpectedStorageWrites();
    // rest of existing code here
}
```

Reply:
done, called it `getAllowedStorageAccess`